lightstep/resource_user_role_binding.go

```diff
@@ -0,0 +1,188 @@
+		DeleteContext: resourceUserRoleBindingDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceUserRoleBindingImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"role": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true, // changing role or project requires a new tf resource to ensure permissions are properly removed.
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true, // changing role or project requires a new tf resource to ensure permissions are properly removed.
+			},
+			"users": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func getUserRoleBindingFromResource(_ context.Context, d *schema.ResourceData) client.RoleBinding {
+	var userRoleBinding client.RoleBinding
+
+	role, ok := d.GetOk("role")
+	if ok {
+		userRoleBinding.RoleName = role.(string)
+	}
+
+	project, ok := d.GetOk("project")
+	if ok {
+		userRoleBinding.ProjectName = project.(string)
+	}
+
+	users, ok := d.GetOk("users")
+	if ok {
+		for _, user := range users.(*schema.Set).List() {
+			userRoleBinding.Users = append(userRoleBinding.Users, user.(string))
+		}
+	}
+
+	return userRoleBinding
+}
+
+func setUserRoleBindingFromResource(d *schema.ResourceData, userRoleBinding client.RoleBinding) error {
+	users := schema.NewSet(schema.HashString, nil)
+	for _, user := range userRoleBinding.Users {
+		users.Add(user)
+	}
+
+	err := d.Set("users", users)
+	if err != nil {
+		return fmt.Errorf("error to set users resource field: %v", err)
+	}
+	err = d.Set("role", userRoleBinding.RoleName)
+	if err != nil {
+		return fmt.Errorf("error to set role resource field: %v", err)
+	}
+	err = d.Set("project", userRoleBinding.ProjectName)
+	if err != nil {
+		return fmt.Errorf("error to set project resource field: %v", err)
+	}
+
+	return nil
+}
+
+func resourceUserRoleCreateOrUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.Client)
+
+	// Read the proposed plan
+	userRoleBinding := getUserRoleBindingFromResource(ctx, d)
+
+	// Update role binding
+	_, err := c.UpdateRoleBinding(ctx, userRoleBinding.ProjectName, userRoleBinding.RoleName, userRoleBinding.Users...)
+	if err != nil {
+		return handleAPIError(err, d, "create/update user role binding")
+	}
+
+	// Save this resource with an ID with the following format "org_name:role_name:project_name",
+	// where project name is optional.
+	d.SetId(fmt.Sprintf("%s/%s", c.OrgName(), userRoleBinding.ID()))
+
+	// update state by forcing a reading to the API.
+	return resourceUserRoleBindingRead(ctx, d, m)
+}
+
+// resourceUserRoleBindingRead reads a user role binding from the resource data.
+//
+// When called by a Read or Delete Context, it will read data from the terraform state.
+// When Called by a Create or Update context, it will reda data from the terraform plan.
```

Comment:
(typos, non-blocking)
```suggestion
// When called by a Create or Update context, it will read data from the terraform plan.
```